servlet/src/main/java/org/togglz/servlet/activation/UrlParameterActivationStrategy.java

```diff
@@ -0,0 +1,117 @@
+package org.togglz.servlet.activation;
+
+import org.togglz.core.activation.ParameterBuilder;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.spi.ActivationStrategy;
+import org.togglz.core.user.FeatureUser;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.togglz.core.activation.Parameter;
+import org.togglz.core.util.Strings;
+import org.togglz.servlet.util.HttpServletRequestHolder;
+import javax.servlet.http.HttpServletRequest;
+
+public class UrlParameterActivationStrategy implements ActivationStrategy {
+	public static final String PARAM_URL_PARAMS = "params";
+	private static final Pattern SPLIT_ON_COMMA = Pattern.compile("\\s*,\\s*");
+	private static final Pattern SPLIT_ON_EQUALS = Pattern.compile("\\s*=\\s*");
+
+	@Override
+	public String getId() {
+		return "url-parameter";
+	}
+
+	@Override
+	public String getName() {
+		return "URL Parameter";
+	}
+
+	@Override
+	public boolean isActive(FeatureState featureState, FeatureUser user) {
+		HttpServletRequest request = HttpServletRequestHolder.get();
+		if (request == null) {
+			return false;
+		}
+
+		String allowedUrlParams = featureState.getParameter(PARAM_URL_PARAMS);
+		if (Strings.isBlank(allowedUrlParams)) {
+			return false;
+		}
+
+		//Parse the request params and activation params and compare them for a match
+		String[] allowedParams = SPLIT_ON_COMMA.split(allowedUrlParams.trim());
+		Map<String, String[]> requestParams = new HashMap<>();
+		requestParams.putAll(request.getParameterMap());
+		String referer = request.getHeader("referer");
+		if (referer != null) {
+			try {
+				Map<String, List<String>> combinedRefererParameters = getRefererParameters(requestParams, referer);
+
+				for (Map.Entry<String, List<String>> entry : combinedRefererParameters.entrySet()) {
+					List<String> val  = entry.getValue();
+					String[] values = new String[val.size()];
+					requestParams.put(entry.getKey(), val.toArray(values));
+				}
+			} catch (URISyntaxException e) {
+				e.printStackTrace();
```

Comment:
Either use a real logger to log this exception or simply ignore it. But add a comment explaining why you ignore it in the latter case.